R/driver-sql-server.R

```diff
@@ -81,6 +81,32 @@ setMethod("dbExistsTable", c("Microsoft SQL Server", "character"),
   }
 )
 
+#' @description
+#' ## `dbListTables()`
+#' The default implementation reports temporary tables as non-existent
+#' when a `catalog_name` isn't supplied since they live in a different catalog.
+#' This method provides a special case for temporary tables.
+#' @rdname SQLServer
+#' @usage NULL
+setMethod("dbListTables", "Microsoft SQL Server",
+  function(conn,
+           catalog_name = NULL,
+           schema_name = NULL,
+           table_name = NULL,
+           table_type = NULL,
+           ...) {
+    res <- callNextMethod()
+
+    if (is.null(catalog_name)) {
+      res_temp <- callNextMethod(conn = conn, catalog_name = "tempdb")
```

Comment:
```suggestion
      res_temp <- callNextMethod(catalog_name = "tempdb")
```

Reply:
Any thoughts on https://github.com/r-dbi/odbc/pull/747#discussion_r1459638726? I see the error noted there with this change.